ipa-core/src/protocol/ipa_prf/malicious_security/verifier.rs

```diff
@@ -104,32 +99,24 @@ pub fn compute_final_sum_share<F: PrimeField, const λ: usize, const P: usize>(z
 ///
 /// The function uses streams since stream offers a chunk method.
 fn recurse_u_or_v<'a, F: PrimeField, J, const λ: usize>(
-    u_or_v_stream: J,
+    u_or_v_iterator: J,
     lagrange_table: &'a LagrangeTable<F, λ, 1>,
-) -> impl Stream<Item = F> + 'a
+) -> impl Iterator<Item = F> + 'a
 where
-    J: Stream<Item = F> + 'a,
+    J: Iterator<Item = F> + 'a,
 {
-    u_or_v_stream.chunks(λ).map(|mut x| {
-        if x.len() < λ {
-            x.extend(repeat(F::ZERO).take(λ - x.len()));
-        }
-        lagrange_table.eval(&x)[0]
-    })
+    u_or_v_iterator
+        .chunk_array::<λ>()
+        .map(|x| lagrange_table.eval(&x)[0])
 }
 
-pub async fn recursively_compute_final_check<
-    F: PrimeField,
-    J,
-    const λ_FIRST: usize,
-    const λ: usize,
->(
-    u_or_v_stream: J,
+pub fn recursively_compute_final_check<F: PrimeField, J, const λ_FIRST: usize, const λ: usize>(
```

Comment:
Consider adding rustdoc explaining the generics here, especially the difference between λ_FIRST and λ